### PR TITLE
[DROOLS-6062] Fix default methods management inside KieAssemblerService

### PR DESCRIPTION
**Pull request by gitgabrio**
@danielezonca @afalhambra  @jiripetrlik

See https://issues.redhat.com/browse/DROOLS-6062

### CODE REVIEW

**Comment by gitgabrio** [comment]
Jenkins run fdb

**Comment by afalhambra** [comment]
Looks good to me - just wait for FDB to turn green. Thanks!

**Review by afalhambra — APPROVED** [review]

**Review by danielezonca — APPROVED** [review]

**Comment by mareknovotny** [comment]
timeout in FDB 
```
[2021-03-04T21:12:18.593Z] [INFO] [talledLocalContainer] 16:08:50,020 WARN  [org.kie.server.services.impl.controller.DefaultRestControllerImpl] (Thread-0 (ActiveMQ-client-global-threads)) No controllers found to update to new status org.kie.server.api.model.KieServerStateInfo@659cc3bc
[2021-03-04T21:12:18.849Z] [INFO] [talledLocalContainer] 16:08:50,289 INFO  [org.kie.server.services.impl.KieServerImpl] (default task-2) Container input-data-string (for release id org.kie.server.testing:input-data-string:1.0.1.Final) successfully stopped
[2021-03-04T21:12:18.849Z] [INFO] [talledLocalContainer] 16:08:50,290 WARN  [org.kie.server.services.impl.controller.DefaultRestControllerImpl] (default task-2) No controllers found to update to new status org.kie.server.api.model.KieServerStateInfo@56884b31
[2021-03-04T21:12:46.974Z] Cannot contact kie-psi-rhel7-xlarge-xmem-36973: java.lang.InterruptedException
[2021-03-05T03:38:16.923Z] Cancelling nested steps due to timeout
[2021-03-05T03:38:16.926Z] Could not connect to kie-psi-rhel7-xlarge-xmem-36973 to send interrupt signal to process
```

jenkins do fdb

File: kie-api/src/main/java/org/kie/api/internal/assembler/KieAssemblerService.java [hunk 1]
```diff
@@ -37,7 +37,9 @@ default void addResourcesBeforeRules(Object kbuilder, Collection<ResourceWithCon
         }
     }
 
-    default void addResourceAfterRules(Object kbuilder, Resource resource, ResourceType type, ResourceConfiguration configuration) throws Exception { }
+    default void addResourceAfterRules(Object kbuilder, Resource resource, ResourceType type, ResourceConfiguration configuration) throws Exception {
+        addResource(kbuilder, resource, type, configuration);
+    }
```
Comment (MarianMacik) [inline kie-api/src/main/java/org/kie/api/internal/assembler/KieAssemblerService.java:42]: Doesn't this introduce a cycle?
* addResource calls addResourceAfterRules
* addResourceAfterRules calls addResource

Comment (gitgabrio) [reply kie-api/src/main/java/org/kie/api/internal/assembler/KieAssemblerService.java:42]: @MarianMacik 
fixed

**Comment by gitgabrio** [comment]
Jenkins run fdb

**Review by MarianMacik — APPROVED** [review]

**Comment by elguardian** [comment]
everybody agree and no dependencies. merged

**Pull request merged**